activerecord/lib/active_record/store.rb

```diff
@@ -143,6 +147,10 @@ def store_accessor(store_attribute, *keys, prefix: nil, suffix: nil)
               read_store_attribute(store_attribute, key)
             end
 
+            define_method("#{accessor_key}?") do
+              read_store_attribute(store_attribute, key).present?
```

Comment:
I don't think this is a good idea, because that's not the same behavior as a real column, making things confusing: https://github.com/rails/rails/blob/5c4a7a711068c1dff2bd873da510ddbb82dcadd7/activerecord/lib/active_record/attribute_methods/query.rb

There was a few other PRs asking for more attribute methods for store accessors, I'm not opposed to it, but we should do some refactoring so we can actually define them like regular attribute methods, and not like some ad hoc thing.

Reply:
@byroot I've investigated replacing these methods with `ActiveRecord::Dirty` and `ActiveModel::AttributeMethods`, but I've hit some challenges.

Are there risks in overriding `attributes` or `attribute_names` for attributes defined by `store` and `store_accessor`? Is it safe for that list to contain attributes that don't map directly to columns or associations?

Reply:
I don't know out of the top of my head, sorry.

That's one of these thing that need quite a bit of investigation.